docs/development/CI/infrastructure.md

```diff
@@ -4,20 +4,111 @@
 
```

Comment:
I would add a short description mentioning that the current implementation of the CI workflows is based on running on the CI service runners the same task automating tools which work locally on developers' workstations like tox and pre-commit, so any test or code checking workflow can be run independently by any developer at any time, without relying on the CI service.